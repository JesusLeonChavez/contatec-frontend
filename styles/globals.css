:root{
  --primary: #706F6F;
  --secondary: #304659;
}

html,
body {
  padding: 0;
  margin: 0;

}

a {
  color: inherit;
  text-decoration: none;
}

* {
  box-sizing: border-box;
  font-family: 'Poppins', sans-serif;
}

.generalWrapper{
  max-width: 1300px;
  margin: 0 auto;
  overflow: auto;
  padding: 0 40px;
}

.flexCenterCustom{
  display: flex;
  align-items: center;
  justify-content: center;
}


/*----------------Swiper-----------------------*/
.swiper-container {
  width: 100%;
  height: 100%;
}

.swiper-slide {
  text-align: center;
  font-size: 18px;
  background: red;
  /* Center slide text vertically */
  display: -webkit-box;
  display: -ms-flexbox;
  display: -webkit-flex;
  display: flex;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  -webkit-align-items: center;
  align-items: center;
}

.slide-2{
  background: green;
}

.mySwiper  .swiper-button-next, .mySwiper  .swiper-button-prev {
  color: white;
  border: 1px solid white;
  border-radius: 50%;
  padding: 1.3rem;
}

.mySwiper .swiper-button-next{
  top: 800px;
  right: 100px;
}

.mySwiper .swiper-button-prev{
  top: 800px;
  right: 180px;
  left: auto;
}

.mySwiper .swiper-button-next::after,.mySwiper .swiper-button-prev::after{
  font-size: 1rem;
}


.mySwiper2{
  width: 200px;
}
.mySwiper2 .swiper-container{
  width: 100%;
  height: 100%;
}
.mySwiper2 .swiper-slide{
  text-align: center;
  font-size: 18px;
  background: #fff;
  background: pink;
  /* Center slide text vertically */
  display: -webkit-box;
  display: -ms-flexbox;
  display: -webkit-flex;
  display: flex;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  -webkit-align-items: center;
  align-items: center;
}
.mySwiper2 .slide-2{
  background: blue;
}


.mySwiper3{
  width:100%;
}
.mySwiper3 .swiper-container{
  width: 300px;
  height: 100%;
}
.mySwiper3 .swiper-slide{
  text-align: center;
  font-size: 18px;
  background: #fff;
  background: purple;
  height: 500px;
  /* Center slide text vertically */
  display: -webkit-box;
  display: -ms-flexbox;
  display: -webkit-flex;
  display: flex;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  -webkit-align-items: center;
  align-items: center;
}
.mySwiper3 .slide-2{
  background: blue;
}

.mySwiper3 .swiper-button-next, .mySwiper3 .swiper-button-prev {
  color: white;
  border: 1px solid white;
  border-radius: 50%;
  padding: 1.3rem;
}

.mySwiper3 .swiper-button-next{
  top: 450px;
  right: 100px;
}
.mySwiper3 .swiper-button-prev{
  top: 450px;
  right: 180px;
  left: auto;
}

.mySwiper3 .swiper-button-next::after, .mySwiper3 .swiper-button-prev::after{
  font-size: 1rem;
}




