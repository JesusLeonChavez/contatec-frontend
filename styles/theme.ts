import { extendTheme } from '@chakra-ui/react'
import { ButtonStyles as Button } from './components/buttonStyles'
export const newTheme = extendTheme({
  colors: {
    primary: '#EB0029',
    secondary: '#FF6F91',
    highlight: '#00C9A7'
  },
  components: {
    Button
  }
})
