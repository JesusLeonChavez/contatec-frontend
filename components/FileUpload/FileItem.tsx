// eslint-disable-next-line @typescript-eslint/ban-ts-comment
// @ts-nocheck
import React from "react"
// eslint-disable-next-line import/no-extraneous-dependencies
import PropTypes from "prop-types"
import styles from "../../styles/components/FileUpload.module.css"
import ZIcon from "../Icon"
import { Consumer } from "./FileUpload"

export default function FileItem({ file, onDelete }) {
  console.log(file)
  return (
    <Consumer>
      {props => (
        <>
          {file.deleted !== true && (
            <div className={styles.fileItemStyled}>
              {
                <>
                  {/* <Icon  name={file.icon} /> */}
                  <ZIcon name="alarm" className="mr1" />
                  <p className="mr-2">{file.name}</p>

                  {!file.error && (
                    <a
                      href={file.url}
                      target="blank"
                      download={!file.name && "prueba.png"}
                      title={!file.name && "prueba.png"}
                    >
                      <ZIcon name="dowload" pointer />
                    </a>
                  )}
                  {file.error && (
                    <span>
                      <ZIcon
                        name="exclamation-circle"
                        pointer
                        // color="var(--danger)"
                        // hover="var(--danger)"
                      />
                    </span>
                  )}
                </>
              }

              {props.remove && !props.readOnly && (
                <ZIcon
                  className="ml2"
                  name="trash"
                  onClick={onDelete}
                  pointer
                />
              )}
            </div>
          )}
        </>
      )}
    </Consumer>
  )
}

FileItem.propTypes = {
  /**
   * file
   */
  file: PropTypes.object.isRequired,
  /**
   * onDelete
   */

  onDelete: PropTypes.func
}
