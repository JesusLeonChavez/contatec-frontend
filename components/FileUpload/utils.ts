/* istanbul ignore file */
/* eslint-disable camelcase */
// eslint-disable-next-line @typescript-eslint/ban-ts-comment
// @ts-nocheck
// eslint-disable-next-line import/no-unresolved
import { FileLink } from "./FileUpload"

export function setFiles(files) {
  // eslint-disable-next-line camelcase
  const new_files = []

  files?.forEach((file, index) => {
    // eslint-disable-next-line no-undef

    if (file instanceof File) {
      new_files.push({
        name: file.name,
        icon: getIconString(file.name),
        url: createFileUrl(file.type, file),
        error: false
      })
    } else if (file instanceof FileLink) {
      new_files.push(file)
    } else {
      if (!file.deleted) {
        new_files.push({
          name: `imagen_${index + 1}`,
          icon: getIconString(file.filename),
          url: file,
          error: false
        })
      }
    }
  })

  return new_files
}

/**

 * @param {String} filename

 * @returns {String}

 */

export function getIconString(filename = ".") {
  const ext = filename.split(".")[1]

  let icon = ""

  switch (ext) {
    case "jpg":
    case "jpeg":
    case "png":
      icon = "image"
      break
    case "docx":
    case "doc":
      icon = "doc"
      break
    case "pdf":
      icon = "pdf"

      break

    default:
      icon = "help-circle"

      break
  }

  return icon
}
export function createFileUrl(type, data) {
  // eslint-disable-next-line no-undef
  return window.URL.createObjectURL(new Blob([data], { type }))
}
