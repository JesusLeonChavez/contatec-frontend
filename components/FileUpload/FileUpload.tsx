/* eslint-disable camelcase */
// eslint-disable-next-line @typescript-eslint/ban-ts-comment
// @ts-nocheck
import React from "react"
// eslint-disable-next-line import/no-extraneous-dependencies
import PropTypes from "prop-types"
import { setFiles, getIconString } from "./utils"
import styles from "../../styles/components/FileUpload.module.css"
import FileItem from "./FileItem"
import ZIcon from "../../components/Icon"
export const { Provider, Consumer } = React.createContext(false)
export function FileLink(
  filename,
  url,
  size = 0,
  extension = "",
  mime_type = ""
) {
  const index = url.indexOf("=")
  const id = url.slice(index)
  this.id = id
  this.name = filename
  this.url = `https://drive.google.com/uc?export=download&id${id}`
  this.delete = false
  this.icon = getIconString(filename)
  this.error = index === -1
  this.extension = extension
  this.size = size
  this.mime_type = mime_type
}

FileLink.propTypes = {
  filename: PropTypes.string,
  id: PropTypes.string
}

type Extensions = "png" | "pdf" | "doc" | "docx" | "jpg" | "jpeg"

interface FileUploadProps {
  files?: any
  readOnly?: boolean
  remove?: boolean
  caption?: string
  placeholder?: string
  outline?: boolean
  disable?: boolean
  onDelete?: (a: number) => void
  onDrop?: (b: any[]) => void
  className?: string
  required?: boolean
  alignment?: "top" | "bottom"
  fullWidth?: boolean
  extensions?: Extensions[]
  style?: Record<string, string>
  loading?: boolean
  errorHelper: boolean
}

export default function FileUpload({
  files,
  extensions,
  readOnly,
  remove,
  caption,
  placeholder,
  outline,
  disabled,
  onDelete,
  onDrop,
  className,
  required,
  alignment,
  fullWidth,
  loading,
  errorHelper,
  ...props
}: FileUploadProps) {
  const [collapse, setCollapse] = React.useState(false)
  const [errorSupport, setErrorSupport] = React.useState(false)
  const toggleFocus = React.useRef(null)
  const handleOut = React.useCallback(
    ev => {
      if (!ev.target.closest("._FileUpload")) {
        setCollapse(false)
      }
    },
    [collapse]
  )

  React.useEffect(() => {
    document
      .querySelector(".chakra-modal__content")
      .addEventListener("click", handleOut)

    return () => {
      document.removeEventListener("click", handleOut)
    }
  }, [handleOut])

  function handleCollapse(value) {
    toggleFocus.current.classList.toggle(styles.focus)
    if (!value) {
      setErrorSupport(false)
    }
    setCollapse(!value)
  }

  function onChange(ev) {
    const _files = ev.target.files
    let availabale_files = []
    let notSupport = false
    let validImage = false
    // eslint-disable-next-line array-callback-return
    availabale_files = Object.values(_files).filter(item => {
      const ext = item.name
        .substring(item.name.lastIndexOf(".") + 1)
        .toLowerCase()
      const size = item.size
      if (extensions.includes(ext)) {
        item.extension = ext
        validImage = true
      } else if (!notSupport) {
        notSupport = true
      }
      if (size <= 1024 * 1024) {
        validImage = true
      } else if (!notSupport) {
        notSupport = true
      }

      if (validImage) {
        return item
      }
      // console.log("gaaa: ", item)
    })

    if (!notSupport) {
      setErrorSupport(false)
      onDrop(Object.values(availabale_files))
    } else {
      setErrorSupport(true)
    }
  }
  const new_files = setFiles(files)
  return (
    <Provider value={{ readOnly, remove }}>
      <div
        className={`_FileUpload ${className} ${
          fullWidth ? styles.fileUploadStyledFullWidth : styles.fileUploadStyled
        } `}
        fullWidth={fullWidth}
      >
        <div
          ref={toggleFocus}
          className={`${styles.fileUploadHead} ${
            errorHelper && styles.errorHelper
          }`}
          outline={outline}
          disabled={disabled}
          required={required}
          onClick={() => handleCollapse(collapse)}
          {...props}
        >
          <i className="fas fa-paperclip mr-4" />
          <div className={styles.containerBadge}>
            <p className="mr2">{placeholder}</p>
            {new_files.length > 0 && (
              <div className={styles.badge}> {new_files.length} </div>
            )}
          </div>
          {/* <i className="fa fa-angle-down mr-1" /> */}
          <ZIcon name="arrow-bottom" className="mr1" />
        </div>
        <div
          className={
            collapse ? styles.fileUploadBodyCollapsed : styles.fileUploadBody
          }
          alignment={alignment}
          collapse={collapse}
          readOnly={readOnly}
        >
          <div className={styles.fileUploadBodyTop}>
            {errorSupport && (
              <p className={styles.textNotSupport}>
                Error: Archivo a cargar no soportado
              </p>
            )}
            <div className={styles.containerFileItems}>
              {!readOnly && new_files.length === 0 && (
                <div className={styles.dragDropStyled}>
                  <i className="fas fa-cloud-upload-alt mb2" />
                  <p className={styles.text}>Arrastra y suelta archivos</p>
                  <p className={`mt2 ${styles.text}`}>
                    Archivos soportados: {extensions?.join(", ")}
                  </p>
                  <input type="file" multiple onChange={onChange} />
                </div>
              )}
              {new_files.map((file, i) => (
                <FileItem key={i} file={file} onDelete={() => onDelete(i)} />
              ))}
              {readOnly && new_files.length === 0 && (
                <div className={styles.caption}>{caption}</div>
              )}
            </div>
          </div>
          {!readOnly && (
            <div className={styles.fileUploadBodyBottom}>
              <div className={styles.containerInputFile}>
                Subir Archivo
                <input
                  className={styles.inputFileStyled}
                  type="file"
                  multiple
                  onChange={onChange}
                />
              </div>
            </div>
          )}
        </div>
      </div>
    </Provider>
  )
}

FileUpload.defaultProps = {
  readOnly: false,
  placeholder: "Archivos adjuntos",
  color: "secondary",
  disabled: false,
  alignment: "bottom",
  caption: "No se han encontrado archivos adjuntos",
  onDrop: () => null,
  outline: true,
  remove: false,
  files: [],
  onDelete: () => null,
  extensions: ["png", "pdf", "doc", "docx", "jpg", "jpeg"],
  loading: false
}
