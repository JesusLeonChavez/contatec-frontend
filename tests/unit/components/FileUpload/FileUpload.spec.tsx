/**

 * @jest-environment jsdom

 */

 import { render } from "@testing-library/react"
 import "@testing-library/jest-dom"
 import { newTheme } from "../../../../styles/theme"
 import { ChakraProvider } from "@chakra-ui/react"
 import FileUpload from "../../../../components/FileUpload/FileUpload"
 
 const ChakraRenderer = ({ children }) => {
   return <ChakraProvider theme={newTheme}>{children}</ChakraProvider>
 }
 
 describe("FileUpload", function () {
  let FileUploadProps
   test("debe renderizarse el componente", () => {
     const component = render(<FileUpload{...FileUploadProps}  />, {
       // eslint-disable-next-line @typescript-eslint/ban-ts-comment
       // @ts-ignore
       wrapper: ChakraRenderer
     })
     expect(component).toMatchSnapshot()
   })
 
 })
 