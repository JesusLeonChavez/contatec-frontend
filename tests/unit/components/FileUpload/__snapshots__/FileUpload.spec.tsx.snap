// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FileUpload debe renderizarse el componente 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body
    class="chakra-ui-light"
  >
    <div
      id="chakra-toast-portal"
    >
      <ul
        id="chakra-toast-manager-top"
        style="position: fixed; z-index: 5500; pointer-events: none; display: flex; flex-direction: column; margin: 0px auto;"
      />
      <ul
        id="chakra-toast-manager-top-left"
        style="position: fixed; z-index: 5500; pointer-events: none; display: flex; flex-direction: column;"
      />
      <ul
        id="chakra-toast-manager-top-right"
        style="position: fixed; z-index: 5500; pointer-events: none; display: flex; flex-direction: column;"
      />
      <ul
        id="chakra-toast-manager-bottom-left"
        style="position: fixed; z-index: 5500; pointer-events: none; display: flex; flex-direction: column;"
      />
      <ul
        id="chakra-toast-manager-bottom"
        style="position: fixed; z-index: 5500; pointer-events: none; display: flex; flex-direction: column; margin: 0px auto;"
      />
      <ul
        id="chakra-toast-manager-bottom-right"
        style="position: fixed; z-index: 5500; pointer-events: none; display: flex; flex-direction: column;"
      />
    </div>
    <div>
      <div
        class="_FileUpload undefined fileUploadStyled "
      >
        <div
          class="fileUploadHead undefined"
          color="secondary"
        >
          <i
            class="fas fa-paperclip mr-4"
          />
          <div
            class="containerBadge"
          >
            <p
              class="mr2"
            >
              Archivos adjuntos
            </p>
          </div>
          <svg
            class="mr1"
            color="var(--primary)"
            fill="none"
            height="1em"
            style="font-size: 15px;"
            viewBox="0 0 11 5"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M10.35.778l-4.8 3.2-4.8-3.2"
              stroke="var(--primary)"
            />
          </svg>
        </div>
        <div
          alignment="bottom"
          class="fileUploadBody"
        >
          <div
            class="fileUploadBodyTop"
          >
            <div
              class="containerFileItems"
            >
              <div
                class="dragDropStyled"
              >
                <i
                  class="fas fa-cloud-upload-alt mb2"
                />
                <p
                  class="text"
                >
                  Arrastra y suelta archivos
                </p>
                <p
                  class="mt2 text"
                >
                  Archivos soportados: 
                  png, pdf, doc, docx, jpg, jpeg
                </p>
                <input
                  multiple=""
                  type="file"
                />
              </div>
            </div>
          </div>
          <div
            class="fileUploadBodyBottom"
          >
            <div
              class="containerInputFile"
            >
              Subir Archivo
              <input
                class="inputFileStyled"
                multiple=""
                type="file"
              />
            </div>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="_FileUpload undefined fileUploadStyled "
    >
      <div
        class="fileUploadHead undefined"
        color="secondary"
      >
        <i
          class="fas fa-paperclip mr-4"
        />
        <div
          class="containerBadge"
        >
          <p
            class="mr2"
          >
            Archivos adjuntos
          </p>
        </div>
        <svg
          class="mr1"
          color="var(--primary)"
          fill="none"
          height="1em"
          style="font-size: 15px;"
          viewBox="0 0 11 5"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M10.35.778l-4.8 3.2-4.8-3.2"
            stroke="var(--primary)"
          />
        </svg>
      </div>
      <div
        alignment="bottom"
        class="fileUploadBody"
      >
        <div
          class="fileUploadBodyTop"
        >
          <div
            class="containerFileItems"
          >
            <div
              class="dragDropStyled"
            >
              <i
                class="fas fa-cloud-upload-alt mb2"
              />
              <p
                class="text"
              >
                Arrastra y suelta archivos
              </p>
              <p
                class="mt2 text"
              >
                Archivos soportados: 
                png, pdf, doc, docx, jpg, jpeg
              </p>
              <input
                multiple=""
                type="file"
              />
            </div>
          </div>
        </div>
        <div
          class="fileUploadBodyBottom"
        >
          <div
            class="containerInputFile"
          >
            Subir Archivo
            <input
              class="inputFileStyled"
              multiple=""
              type="file"
            />
          </div>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
